catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1ResourceBase.java

```diff
@@ -259,53 +255,18 @@ static Branch checkBranch(Reference reference) {
     return (Branch) reference;
   }
 
-  protected String defaultTableLocation(String location, TableRef tableRef) {
-    if (location != null) {
-      return location;
-    }
-
-    WarehouseConfig warehouse = catalogConfig.getWarehouse(tableRef.warehouse());
-    ContentKey key = tableRef.contentKey();
+  protected String defaultTableLocation(TableRef tableRef) {
+    return defaultTableLocation(tableRef.warehouse(), tableRef.contentKey());
+  }
 
+  protected String defaultTableLocation(String warehouseName, ContentKey key) {
+    WarehouseConfig warehouse = catalogConfig.getWarehouse(warehouseName);
+    String baseLocation = warehouse.location();
     Namespace ns = key.getNamespace();
     if (!ns.isEmpty()) {
-      List<ContentKey> parentNamespaces = new ArrayList<>();
-      for (Namespace namespace = key.getNamespace();
-          !namespace.isEmpty();
-          namespace = namespace.getParentOrEmpty()) {
-        parentNamespaces.add(namespace.toContentKey());
-      }
-
-      String baseLocation = concatLocation(warehouse.location(), ns.toString());
-      try {
-        ParsedReference parsedRef = requireNonNull(tableRef.reference());
-        GetMultipleContentsResponse namespacesResp =
-            nessieApi
-                .getContent()
-                .reference(Branch.of(parsedRef.name(), parsedRef.hashWithRelativeSpec()))
-                .keys(parentNamespaces)
-                .getWithResponse();
-        Map<ContentKey, Content> namespacesMap = namespacesResp.toContentsMap();
-        for (ContentKey nsKey : parentNamespaces) {
-          Content namespace = namespacesMap.get(nsKey);
-          if (namespace instanceof Namespace) {
-            String namespaceLocation = ((Namespace) namespace).getProperties().get("location");
-            if (namespaceLocation != null) {
-              baseLocation = namespaceLocation;
-              break;
-            }
-          }
-        }
-      } catch (NessieNamespaceNotFoundException e) {
-        // ignore
-      } catch (NessieNotFoundException e) {
-        // do nothing we want the same behavior that if the location is not defined
-      }
-
-      location = concatLocation(baseLocation, key.getName());
-    } else {
-      location = concatLocation(warehouse.location(), key.getName());
+      baseLocation = concatLocation(warehouse.location(), ns.toString());
```

Comment:
`ns.toString()` will contain special chars (`\u0000`) if it has multiple elements.

Reply:
Given https://github.com/apache/iceberg/issues/10279 I'd be very cautious with constructing storage paths from user-provided data. Nessie keys are pretty liberal and permit all sorts of special chars. I think we ought to weed them out either by replacing with a placeholder or encoding (unidirectional is ok, IMHO) with non-special chars.

Reply:
Putting -1 on this just in case :)

Reply:
I noticed. I didn't change that though, the `ns.toString()` call was there already. Will look into an alternative.

Reply:
Yeah, but we only recently discovered the Iceberg-side processing issues with special chars :facepalm: I think it would be wise to avoid any special chars in the locations we control at least.

Reply:
I'd be fine with doing this in a follow-up PR too.

Reply:
Indeed I think this will have to go in a separate PR because it will likely modify `org.projectnessie.model.Namespace` and add a `toSafePathString` method or stg in the style.

Reply:
SGTM

Reply:
#8524